pkg/awsclient/client.go

```diff
@@ -78,6 +79,52 @@ func (c *awsClient) ListResourceRecordSets(input *route53.ListResourceRecordSets
 	return c.route53Client.ListResourceRecordSets(input)
 }
 
+// SearchForHostedZone finds a hostedzone when given an aws client and a domain string
+// Returns a hostedzone object
+func SearchForHostedZone(r53svc Client, baseDomain string) (hostedZone route53.HostedZone, err error) {
+	hostedZoneOutput, err := r53svc.ListHostedZones(&route53.ListHostedZonesInput{})
+	if err != nil {
+		return hostedZone, err
+	}
+
+	for _, zone := range hostedZoneOutput.HostedZones {
+		if strings.EqualFold(baseDomain, *zone.Name) && !*zone.Config.PrivateZone {
+			hostedZone = *zone
+		}
+	}
+	return hostedZone, err
+}
+
+// BuildR53Input contructs an Input object for a hostedzone. Contains no recordsets.
+func BuildR53Input(hostedZone string) *route53.ChangeResourceRecordSetsInput {
+	input := &route53.ChangeResourceRecordSetsInput{
+		ChangeBatch: &route53.ChangeBatch{
+			Changes: []*route53.Change{},
+		},
+		HostedZoneId: &hostedZone,
+	}
+	return input
+}
+
+// CreateR53TXTRecordChange creates an route53 Change object for a TXT record with a given name
+// and a given action to take.
+func CreateR53TXTRecordChange(name *string, action string, value *string) route53.Change {
+	change := route53.Change{
+		Action: aws.String(route53.ChangeActionDelete),
```

Comment:
I see `action` passed in, but not used. Should it go on this line instead of `ChangeActionDelete`?

Reply:
Good catch. It looks like I forgot to write the switch/case statement for setting dynamic actions.

Reply:
Updated, now checks the action passed and sets the correct r53 action.